---
title: "Single-cell RNAseq of ZFP541 manuscript (10x)"
author: "Jarning"
date: '`r format(Sys.time(), "%d %B, %Y")`'
output: html_document
---

```{r setup}
options(stringsAsFactors = F)
library(Seurat)
library(SeuratWrappers)
library(tidyverse)
```

### Batch correction

```{r loadData}
seu <- readRDS("tmp/01.seu.QC.rds")
```

```{r norm and hvgs}
seu.list <- SplitObject(seu, split.by = "orig.ident")##############separate samples
for (i in 1:length(seu.list)) {#################normalize data
  seu.list[[i]] <- NormalizeData(seu.list[[i]], verbose = FALSE)
  seu.list[[i]] <- FindVariableFeatures(seu.list[[i]], selection.method = "vst", 
                                        nfeatures = 2000, verbose = FALSE)############### return the 2000 most highly variable genes
}
```

```{r mnnCorrect and umap, fig.width=8, fig.height=5}
seu <- RunFastMNN(object.list = seu.list)#######Correct for batch effects in single-cell expression data using a fast version of the mutual nearest neighbors (MNN) method
seu <- RunUMAP(seu, reduction = "mnn", dims = 1:30)#########run the Uniform Manifold Approximation and Projection (UMAP) dimensional reduction technique. 
DimPlot(seu, split.by = "orig.ident") + NoLegend()###########Graphs the output of a dimensional reduction technique on a 2D scatter plot where each point is a cell and it's positioned based on the cell embeddings determined by the reduction technique. 
```

```{r markers, fig.width=8, fig.height=6}
FeaturePlot(seu, features = c("Dazl","Vim"), pt.size = .5, split.by = "orig.ident", ncol = 2)       # germ cells vs somatic cells Colors single cells on a dimensional reduction plot according to a 'feature' (i.e. gene expression, PC scores, number of genes detected, etc.)
FeaturePlot(seu, features = c("Gfra1","Ccnd2"), pt.size = .5, split.by = "orig.ident", ncol = 2)    # Spermatogonia
FeaturePlot(seu, features = c("Stra8","Gm4969"), pt.size = .5, split.by = "orig.ident", ncol = 2)   # pre leptotene spermatocytes
FeaturePlot(seu, features = c("Zcwpw1","Prdm9"), pt.size = .5, split.by = "orig.ident", ncol = 2)   # leptotene spermatocytes
FeaturePlot(seu, features = c("Meiob","Rad51ap2"), pt.size = .5, split.by = "orig.ident", ncol = 2) # zygotene  spermatocytes
FeaturePlot(seu, features = c("Piwil1","Mybl1"), pt.size = .5, split.by = "orig.ident", ncol = 2)   # pachytene spermatocytes
FeaturePlot(seu, features = c("Pou5f2","Ccna1"), pt.size = .5, split.by = "orig.ident", ncol = 2)   # diplotene spermatocytes
FeaturePlot(seu, features = c("Sun5","Tex36"), pt.size = .5, split.by = "orig.ident", ncol = 2)     # round spermatids
FeaturePlot(seu, features = c("Prm1","Cstl1"), pt.size = .5, split.by = "orig.ident", ncol = 2)     # enlongated spermatids
```

```{r saveData}
saveRDS(seu, "tmp/02.seu.QC.batchCorrect.rds")
```

```{r info}
sessionInfo()
```

